File: 1566-check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence/1566-check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence.go
func isPrefix(word string, sWord string) bool {
  if len(sWord) > len(word) {
    return false
  }
    
  for i := 0; i < len(sWord); i++ {
    if word[i] != sWord[i] {
      return false
    }
  }

  return true

}

func isPrefixOfWord(sentence string, searchWord string) int {
    words := strings.Split(sentence," ")

    for i := 0; i < len(words);i++ {
      if isPrefix(words[i],searchWord){
        return i+1
      }
    }

    return -1
}

File: 1566-check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence/1566-check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence.ts
function isPrefix(word : string, sWord : string) : boolean{
  if(sWord.length > word.length) return false

  for(let i = 0;i < sWord.length;i++){
    if(sWord[i] !== word[i]){
      return false
    }
  }

  return true
}

function isPrefixOfWord(sentence: string, searchWord: string): number {
    let words = sentence.split(" ")

    for(let i = 0;i < words.length;i++){
      if(isPrefix(words[i],searchWord)){
        return i+1
      }
    }

    return -1
};